src/sagemaker/cli/compatibility/v2/modifiers/tf_legacy_mode.py

```diff
@@ -0,0 +1,150 @@
+        )
+
+        is_in_tensorflow_module = (
+            isinstance(node.func.value, ast.Attribute)
+            and node.func.value.attr == "tensorflow"
+            and isinstance(node.func.value.value, ast.Name)
+            and node.func.value.value.id == "sagemaker"
+        )
+
+        return ends_with_tensorflow_constructor and is_in_tensorflow_module
+
+    def _is_legacy_mode(self, node):
+        """Checks if the ``ast.Call`` node's keywords signal using legacy mode."""
+        script_mode = False
+        py_version = "py2"
+
+        for kw in node.keywords:
+            if kw.arg == "script_mode":
+                script_mode = bool(kw.value.value)
+            if kw.arg == "py_version":
+                py_version = kw.value.s
+
+        return not (py_version.startswith("py3") or script_mode)
+
+    def modify_node(self, node):
+        """Modifies the ``ast.Call`` node's keywords to turn TensorFlow legacy mode parameters
+        into hyperparameters and set ``script_mode=False``.
+
+        The parameters that are converted into hyperparameters:
+
+        - ``training_steps``
+        - ``evaluation_steps``
+        - ``checkpoint_path``
+        - ``requirements_file``
+
+        Args:
+            node (ast.Call): a node that represents a TensorFlow constructor.
+        """
+        base_hps = {}
+        additional_hps = {}
+        kw_to_remove = []  # remove keyword args after so that none are skipped during iteration
+
+        for kw in node.keywords:
+            if kw.arg == "script_mode":
+                # remove here because is set to False later regardless of current value
+                kw_to_remove.append(kw)
+            if kw.arg == "hyperparameters" and kw.value:
+                base_hps = dict(zip(kw.value.keys, kw.value.values))
+                kw_to_remove.append(kw)
+            if kw.arg in self.LEGACY_MODE_PARAMETERS and kw.value:
+                hp_key = self._hyperparameter_key_for_param(kw.arg)
+                additional_hps[hp_key] = kw.value
+                kw_to_remove.append(kw)
+
+        self._remove_keywords(node, kw_to_remove)
+        self._add_updated_hyperparameters(node, base_hps, additional_hps)
+
+        node.keywords.append(ast.keyword(arg="script_mode", value=ast.NameConstant(value=False)))
+
+    def _hyperparameter_key_for_param(self, arg):
+        """Returns an ``ast.Str`` for a hyperparameter key replacing a legacy mode parameter."""
+        name = "sagemaker_requirements" if arg == "requirements_file" else arg
+        return ast.Str(s=name)
+
+    def _remove_keywords(self, node, keywords):
+        """Removes the keywords from the ``ast.Call`` node."""
+        for kw in keywords:
+            node.keywords.remove(kw)
+
+    def _add_updated_hyperparameters(self, node, base_hps, additional_hps):
+        """Combines and adds the hyperparameters to the ``ast.Call`` node's keywords."""
+        base_hps.update(additional_hps)
```

Comment:
I don't see any chance of collision (which would favor `LEGACY_MODE_PARAMETERS`)

Reply:
it's possible that someone could both set `checkpoint_path` in their constructor and have it in their manually-defined hyperparameters - currently, if someone does that, the parameter is favored ([source](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/tensorflow/estimator.py#L709-L716))